nautobot/users/factory.py

```diff
@@ -13,27 +13,9 @@
 class UserFactory(BaseModelFactory):
     class Meta:
         model = User
-        exclude = ("has_first_name", "has_last_name", "has_email")
-
-    has_first_name = NautobotBoolIterator()
-    first_name = factory.Maybe("has_first_name", factory.Faker("first_name"), "")
-
-    has_last_name = NautobotBoolIterator()
-    last_name = factory.Maybe("has_last_name", factory.Faker("last_name"), "")
-
-    username = factory.Maybe(
-        "has_first_name",
-        factory.Maybe(
-            "has_last_name",
-            factory.LazyAttribute(lambda u: f"{u.first_name[0].lower()}{u.last_name.lower()}"),
-            factory.LazyAttribute(lambda u: u.first_name.lower()),
-        ),
-        factory.Maybe(
-            "has_last_name",
-            factory.LazyAttribute(lambda u: u.last_name.lower()),
-            factory.Faker("user_name"),
-        ),
-    )
+        exclude = "has_email"
+
+    username = factory.Faker("user_name")
```

Comment:
Why this change? Removing the first_name/last_name properties from the generated user feels like a step backwards to me.

Reply:
Yeah, I think our old way of constructing a username was flaky and it was running into duplicate name issues in MySQL https://github.com/nautobot/nautobot/actions/runs/7149742147/job/19472323800. So I thought it might be more straightforward to just generate a user_name directly.

Reply:
Would be interesting to know what name the faker was running into issues with. I wonder if just changing it so that first_name and last_name were always present instead of randomly absent would be a cleaner fix. Not critical though.